### ISSUE TITLE
False positives if local workgroup size is set via specialization constant

### ISSUE THREAD

**Issue by hawk1x** [KhronosGroup/Vulkan-ValidationLayers#10057]
**Environment:**
 - OS: windows 10
 - GPU and driver version: nvidia gtx950 576.02
 - SDK or header version if building from repo: 1.4.309.0
 - Options enabled (synchronization, best practices, etc.): synchronization, best practices amd & arm

**Describe the Issue**

False positives if local workgroup size is set via specialization constant.

**Expected behavior**

--

**Valid Usage ID**

```sh
Validation Performance Warning: [ BestPractices-AMD-LocalWorkgroup-Multiple64 ] | MessageID = 0x85f09c61
vkCreateComputePipelines(): pCreateInfos[0] [AMD] compute shader with work group dimensions (1, 1, 1), workgroup size (1), is not a multiple of 64. Make the workgroup size a multiple of 64 to obtain best performance across all AMD GPU generations.

Validation Performance Warning: [ BestPractices-Arm-vkCreateComputePipelines-compute-thread-group-alignment ] | MessageID = 0x79ec8938
vkCreateComputePipelines(): pCreateInfos[0] [Arm] compute shader with work group dimensions (1, 1, 1) is not aligned to 4 threads. On Arm Mali architectures, not aligning work group sizes to 4 may leave threads idle on the shader core.

Validation Performance Warning: [ BestPractices-Arm-vkCreateComputePipelines-compute-spatial-locality ] | MessageID = 0x6d8fa932
vkCreateComputePipelines(): pCreateInfos[0] [Arm] compute shader has work group dimensions (1, 1, 1), which suggests a 1D dispatch, but the shader is accessing 2D or 3D images. The shader may be exhibiting poor spatial locality with respect to one or more shader resources.
```

**Additional context**

--

**Comment by spencer-lunarg** [comment]
ok, so this one we also had to do a lot of work around logic in CoreChecks where we handle the fact this can be a spec constant... seems Best Practice doesn't have that (nor currently an easy way to do it)

This will require a bit more work to plumb out how to route the Spec Constant for everyone (both Core Checks and Best Practice) to have

Might need a way to tie this info to the pipeline/shaderObject state itself
